### DISCUSSION TITLE
Parcel2 nightly build lit-element project for modern browsers only

### DISCUSSION THREAD

**Discussion by TomzBench** [parcel-bundler/parcel#5970] [Q&A] [answered]
Hello,

I am trying to get parcel to bundle my lit-element project to target modern browsers only. Parcel seems to be transpiling my code too far and i'm not sure how to output the javascript to be more modern.

I created a small repository with my attempt located here: https://github.com/TomzBench/parcel-lit

I followed some steps to try and get parcel working with lit-element here: https://github.com/parcel-bundler/parcel/issues/2919

I get the same error from issue 2919:  `Uncaught TypeError: HTMLElement constructor: 'new' is required`

In general, what is the best way to compile lit element with Parcel?  

Also (a second question) - when invoking parcel programmatically via the Parcel API, I invoke pointing to a .parcelrc file. But how does Parcel know where the .babelrc is if I am in a monorepo? Does it assume it is next to .parcelrc?

**Comment by mischnic** [comment, marked as answer]
1. Remove `"main": "index.html"` from package.json
2. Just use this as the parcelrc
```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.{js,mjs,jsm,jsx,es6,cjs,ts,tsx}": [
      "@parcel/transformer-typescript-tsc",
      "@parcel/transformer-babel",
      "@parcel/transformer-js"
    ]
  }
}
```


3. The actual problem here is that your tsconfig is only applied to your source files, not to node_modules. So they are transpiled with `"target": "es3"` (default for tsc). A workaround is only transpiling your source files with tsc:

```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "**/src/**/*.{js,mjs,jsm,jsx,es6,cjs,ts,tsx}": [
      "@parcel/transformer-typescript-tsc",
      "@parcel/transformer-babel",
      "@parcel/transformer-js"
    ]
  }
}
```
(`**/src/**/*.` because this matching is somewhat broken at the moment, I'll need to change it so that it's actually a relative path...)


> Parcel know where the .babelrc is if I am in a monorepo? Does it assume it is next to .parcelrc?

There are special rules for monorepos for Babel: https://babeljs.io/docs/en/config-files#monorepos
(Parcel doesn't do anything special here)

**Reply by TomzBench** [reply]
Thanks. Working now. Much appreciated

**Comment by TomzBench** [comment]
@mischnic

This also fixed my monorepo. However, I have to build my project twice, as I get random typescript compiler errors that heal them selves on he second build. I'll see if I can extend my small repo and get it to reproduce there.
